Support child projects android

codingjeremy opened the pull request:
Includes both shell (previously reviewed) and now python side to update dependencies.

The python code works, but I'm not a python expert. I did have someone who knows python look over parts of the changes, but I would like a deep review to make sure it looks/reads ok.

Please note, Android basically moved support libs to android x, so I added 'androidx' to GROUPS_ALLOWED_MAJOR_UPDATE. Let me know if that is too general. I don't think it is, as we want our samples to use the latest major updates or break (so we can fix).

samtstern commented on fix_android_dependencies.py:
```diff
@@ -107,5 +107,35 @@ def update_all():
+    project_root = os.getcwd()
+    print 'Repo root: {}'.format(project_root)
+
+    top_level_report = project_root + RELATIVE_PATH_TO_JSON_REPORT
```
For adding paths together we should use `os.path.join()` instead of string operations.

samtstern commented:
One small comment but LGTM.

samtstern commented:
@codingjeremy just to be clear, all of the dependency reporting functionality depends on this plugin:
https://github.com/ben-manes/gradle-versions-plugin

So to use it on Android repos you will have to configure that plugin.  I know Android is pretty strict about third-party dependencies.

samtstern commented:
@codingjeremy probably out of scope but it looks like we could replace almost all of our custom code and just use this thing:
https://github.com/patrikerdes/gradle-use-latest-versions-plugin

codingjeremy commented:
Thanks, @samtstern!  I addressed the comment.

I will make sure I add ben manes' plugin info to the projects. 

Thanks for finding that other plugin. I will look into it. Might need to followup later with another PR.

samtstern commented:
@codingjeremy thanks!  LGTM
